### ISSUE TITLE
A crash

### ISSUE THREAD

**Issue by weiyixuan** [influxdata/tdigest#7]
panic: runtime error: index out of range

goroutine 20 [running]:
github.com/influxdata/tdigest.(*TDigest).Quantile(0xc420133380, 0x3fd0000000000000, 0x3ff0000000000000)
	/home/cc/gocode/src/github.com/influxdata/tdigest/tdigest.go:135 +0x277

**Comment by goller** [comment]
Hi @weiyixuan , thanks for writing in!  Do you happen to have the input data that caused this?

**Comment by weiyixuan** [comment]
This problem is hard to reproduce. I've been testing for a long time and still haven't reappeared.

**Comment by goller** [comment]
@weiyixuan Ok, I'll try to create some tests cases and figure out how to reproduce.

**Comment by goller** [comment]
@weiyixuan I can now reproduce your panic with a test case using your data!

```
goroutine 16 [running]:
testing.tRunner.func1(0xc420058a50)
	/usr/local/Cellar/go/1.10.3/libexec/src/testing/testing.go:742 +0x29d
panic(0x122d3a0, 0x137f8e0)
	/usr/local/Cellar/go/1.10.3/libexec/src/runtime/panic.go:502 +0x229
github.com/influxdata/tdigest.(*TDigest).Quantile(0xc42004c580, 0x3fd0000000000000, 0x3ff0000000000000)
	/Users/goller/src/go/src/github.com/influxdata/tdigest/tdigest.go:135 +0x277
github.com/influxdata/tdigest_test.TestTdigest_Quantile.func1(0xc420058a50)
	/Users/goller/src/go/src/github.com/influxdata/tdigest/tdigest_test.go:127 +0xd9
testing.tRunner(0xc420058a50, 0xc42007a670)
	/usr/local/Cellar/go/1.10.3/libexec/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.10.3/libexec/src/testing/testing.go:824 +0x2e0
FAIL	github.com/influxdata/tdigest	0.365s```

I'll work on a fix.

**Comment by goller** [comment]
@weiyixuan I have a fix ready, but, my access to the repo was removed accidently.

If you would like to try now before I merge the PR you need to edit this line:

https://github.com/influxdata/tdigest/blob/master/tdigest.go#L126

It needs to be:

```go
	if index <= t.processed[0].Weight/2.0 {
		return t.min + 2.0*index/t.processed[0].Weight*(t.processed[0].Mean-t.min)
}
```

I've changed it from `index < ` to `index <=`

**Comment by weiyixuan** [comment]
Thanks!

**Comment by goller** [comment]
Hi @weiyixuan, this is fixed by #8 and has been merged to master.  Do you mind trying it out?

**Comment by weiyixuan** [comment]
No problem yet
